ui/src/app/job-list/job-list.component.ts

```diff
@@ -67,8 +67,11 @@ export class JobListComponent implements OnInit {
         this.reloadJobs(this.route.snapshot.queryParams['q'], true);
       }
     });
+    const req = URLSearchParamsUtils.unpackURLSearchParams(this.route.snapshot.queryParams['q']);
+    this.projectId = req.extensions.projectId || '';
 
     this.header.pageSubject.subscribe(resp => this.onClientPaginate(resp));
+    this.pageSize = this.settingsService.getPageSize(this.projectId) || 50;
```

Comment:
If you're going to leave it as an `@Input`, you should keep the initialization to 50 as it was before, and then only override that value here if getPageSize() is truthy.

Reply:
done